lib/ts/station/station.ex

```diff
@@ -22,20 +25,24 @@ defmodule Station do
   end
 
   def stop(pid) do
+    Logger.info(fn -> "Station stopped at pid=#{inspect pid}" end)
     GenServer.stop(pid, :normal)
   end
 
   # Getting the current schedule
   def get_timetable(pid) do
+    Logger.info(fn -> "Getting the current schedule at pid=#{inspect pid}" end)
     GenServer.call(pid, :get_schedule)
   end
 
   # Updating the current state
   def update(pid, new_vars) do
+    Logger.info(fn -> "Updating the current state at pid=#{inspect pid}" end)
     GenServer.cast(pid, {:update, new_vars})
   end
 
   def send_query(pid, query) do
+    Logger.info(fn -> "Sending query to pid=#{inspect pid}" end)
```

Comment:
Despite its name, this function helps receive queries from neighbours. Hence the log message must be "Received query <query_id> at station <station_name/id>."